src/helpers.rs

```diff
@@ -954,17 +953,17 @@ impl<'mir, 'tcx> MiriMachine<'mir, 'tcx> {
     pub fn caller_span(&self) -> Span {
         // We need to go down at least to the caller (len - 2), or however
         // far we have to go to find a frame in a local crate which is also not #[track_caller].
-        let frame_idx = self.top_user_relevant_frame();
-        let stack = self.stack();
-        let frame_idx = cmp::min(frame_idx, stack.len().saturating_sub(2));
-        stack.get(frame_idx).map(Frame::current_span).unwrap_or(rustc_span::DUMMY_SP)
+        self.top_user_relevant_frame()
+            .map(|frame_idx| cmp::min(frame_idx, self.stack().len() - 2))
```

Comment:
This changes semantics I think?
```suggestion
            .map(|frame_idx| cmp::min(frame_idx, self.stack().len().saturating_sub(2)))
```

Reply:
Does it make sense to ask for the caller span (especially in an FnEntry retag) when we have one stack frame? I wrote this thinking "no, that would be a bug, so a crash would be good"

Reply:
then it should be `checked_sub(2).unwrap()`